docs/send-finality-vote.md

```diff
@@ -0,0 +1,90 @@
+# Finality Votes Submission Specification
+
+## Overview
+
+Finality providers submit votes to finalize blocks on the consumer chain.
+This document specifies the process of submitting finality votes.
+
+## Internal State
+
+The finality provider maintains two critical persistent states:
+
+- Last voted height
+- Last processed height
+
+### Last voted height
+
+Tracks the most recent height for which a finality vote was successfully
+submitted. This prevents duplicate voting on previously voted heights.
+
+### Last processed height
+
+Tracks the most recent height for which a voting decision was made
+(whether the decision was to vote or not).
+By definition, `LastProcessedHeight >= LastVotedHeight` always holds.
+
+## Submission Process
+
+### Bootstrapping
+
+To determine the initial processing height:
+
+1. Query consumer chain for `lastFinalizedHeight` (`0` if no finalized blocks)
+2. Query consumer chain for `finalityActivationHeight`:
+3. Query `lastVotedHeightRemote` from consumer chain (`0` if no votes are sent)
+4. Synchronize local state:
+   - Ensure local `lastVotedHeight` and `lastProcessedHeight` ≥ `lastVotedHeightRemote`
+5. Begin processing at:
+   - `max(lastProcessedHeight + 1, lastFinalizedHeight + 1, finalityActivationHeight)`
```

Comment:
two things:
- `lastProcessedHeight` is not initialised anywhere. I assume it comes from persistent state. Lets add the step of retrievieng it from database
- in some consumer chains, it maybe possible to vote and receive reward even after finality is reached, so starting from `lastFinalizedHeight + 1` may leave some rewards on the table. I would maybe leave it as comment here.

Reply:
> lastProcessedHeight is not initialised anywhere. I assume it comes from persistent state. Lets add the step of retrievieng it from database

In the discussion [here](https://github.com/babylonlabs-io/finality-provider/pull/120#discussion_r1836699698) I think we could probably be able to remove `lastProcessedHeight` as it is not critical to security.

> in some consumer chains, it maybe possible to vote and receive reward even after finality is reached, so starting from lastFinalizedHeight + 1 may leave some rewards on the table. I would maybe leave it as comment here.

Good point. Fixed in [8732f15](https://github.com/babylonlabs-io/finality-provider/pull/120/commits/8732f15c540d9f4361b9dace934cf4a87aebb3e6)